src/cdk-experimental/menu/menu-stack.spec.ts

```diff
@@ -0,0 +1,108 @@
+import {QueryList, ViewChild, ViewChildren, Component} from '@angular/core';
+import {CdkMenu} from './menu';
+import {CdkMenuBar} from './menu-bar';
+import {ComponentFixture, TestBed, async} from '@angular/core/testing';
+import {CdkMenuItemTrigger} from './menu-item-trigger';
+import {MenuStack} from './menu-stack';
+import {CdkMenuModule} from './menu-module';
+
+describe('MenuStack', () => {
+  let fixture: ComponentFixture<MultiMenuWithSubmenu>;
+  let menuStack: MenuStack;
+  let triggers: CdkMenuItemTrigger[];
+  let menus: CdkMenu[];
+
+  /** Fetch triggers, menus and the menu stack from the test component.  */
+  function getElementsForTesting() {
+    fixture.detectChanges();
+    triggers = fixture.componentInstance.triggers.toArray();
+    menus = fixture.componentInstance.menus.toArray();
+    menuStack = fixture.componentInstance.menuBar._menuStack;
+  }
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      imports: [CdkMenuModule],
+      declarations: [MultiMenuWithSubmenu],
+    }).compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(MultiMenuWithSubmenu);
+    fixture.detectChanges();
+
+    getElementsForTesting();
+  });
+
+  /** Open up all of the menus in the test component. */
+  function openAllMenus() {
+    triggers[0].openMenu();
+    getElementsForTesting();
+    triggers[1].openMenu();
+    getElementsForTesting();
+    triggers[2].openMenu();
+    getElementsForTesting();
+  }
+
+  it(
+    'should fill the menu stack with the latest menu at the end of the stack and oldest at' +
+      ' the start of the stack',
+    () => {
+      openAllMenus();
+      expect(menus.length).toBe(3);
+
+      menuStack.close.subscribe(item => {
```

Comment:
Having an `expect` in a subscribe is tricky because you're not guaranteed that the stream will actually emit.